client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaUtils.java

```diff
@@ -314,6 +318,58 @@ public static DatumWriter<?> getDatumWriter(
     }
   }
 
+  public static JsonNode findMatchingField(JsonNode node, String path) {
+    String [] identifiers = path.split("\\.");
+    if (identifiers.length < 2) {
+      throw new IllegalArgumentException(
+        String.format("Given field path '%s' has less than two components.", path));
+    }
+
+    // ignore namespace since recordName and fieldName combination is unique in a schema
+    String fieldName = identifiers[identifiers.length - 1];
+    String recordName = identifiers[identifiers.length - 2];
+
+    LinkedList<JsonNode> toVisit = new LinkedList<>();
+    toVisit.add(node);
+
+    // BFS to search field
+    while (toVisit.size() > 0) {
+      JsonNode curr = toVisit.removeFirst();
+      if (!curr.has("type")) {
+        // union
+        curr.elements().forEachRemaining(toVisit::add);
+      } else {
+        String type = curr.get("type").asText();
+        switch (type) {
+          case "record":
+            if (recordName.equals(curr.get(NAME_FIELD).asText())) {
```

Comment:
I think you'll want to check the namespace as well

Reply:
Fixed.